src/ensembl/src/content/app/browser/browser-region-editor/BrowserRegionEditor.tsx

```diff
@@ -0,0 +1,336 @@
+    const params = {
+      genomeId: props.activeGenomeId,
+      focus: activeEnsObjectId,
+      location: chrLocationStr
+    };
+
+    props.replace(urlFor.browser(params));
+  };
+
+  const resetForm = () => {
+    updateErrorMessages(null, null);
+
+    props.toggleRegionEditorActive(false);
+    props.resetRegionValidation();
+  };
+
+  const updateBrowser = () => {
+    const newChrLocation: ChrLocation = [
+      chrInput,
+      getNumberWithoutCommas(locationStartInput),
+      getNumberWithoutCommas(locationEndInput)
+    ];
+
+    if (chrInput === region) {
+      changeLocation(newChrLocation);
+    } else {
+      changeFocusObject(newChrLocation);
+    }
+  };
+
+  const closeForm = () => {
+    updateAllInputs();
+    resetForm();
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const region = `${chrInput}:${locationStartInput}-${locationEndInput}`;
+    props.validateRegion(region);
+  };
+
+  useEffect(() => {
+    updateAllInputs();
+  }, [props.chrLocation]);
+
+  useEffect(() => () => props.resetRegionValidation(), []);
+
+  useEffect(() => {
+    if (props.isActive) {
+      const { validationInfo, isValidationInfoLoading } = props;
+
+      if (!isValidationInfoLoading) {
+        const errorMessages = getRegionEditorErrorMessages(validationInfo);
+
+        if (
+          errorMessages.locationStartError ||
+          errorMessages.locationEndError
+        ) {
+          updateErrorMessages(
+            errorMessages.locationStartError,
+            errorMessages.locationEndError
+          );
+        } else {
+          resetForm();
+          updateBrowser();
+        }
+      }
+    }
+  }, [props.validationInfo]);
+
+  const locationStartRef = useRef<HTMLDivElement>(null);
+  const locationEndRef = useRef<HTMLDivElement>(null);
+
+  const regionEditorClassNames = classNames(styles.browserRegionEditor, {
+    [browserStyles.semiOpaque]: props.isDisabled
+  });
+
+  const locationStartClassNames = classNames({
+    [browserNavBarStyles.errorText]: locationStartErrorMessage
+  });
+
+  const locationEndClassNames = classNames({
+    [browserNavBarStyles.errorText]: locationEndErrorMessage
+  });
+
+  const buttonsClassNames = classNames(
+    browserNavBarStyles.browserNavBarButtons,
+    {
+      [browserNavBarStyles.browserNavBarButtonsVisible]: props.isActive
+    }
+  );
+
+  return (
+    <div className={regionEditorClassNames}>
+      {props.isDisabled ? (
+        <div className={browserStyles.browserOverlay}></div>
+      ) : null}
+      <form onSubmit={handleSubmit} onFocus={handleFocus}>
+        <div className={styles.inputGroup}>
+          <label className="show-for-large">Chr</label>
+          <Select
+            onSelect={updateRegionInput}
+            options={getKaryotypeOptions()}
+          ></Select>
+        </div>
+        <div
+          className={styles.inputGroup}
+          role="startInputGroup"
```

Comment:
Now, here we actually have a class (styles.inputGroup) that will let us target this div in the test. Why is it targeted by its role then?

(Same below)

Reply:
It's because it is good to have some kind of identifier for the actual set of elements that we are testing?

Reply:
Why doesn't the `. inputGroup` class serve as an identifier?

Reply:
It is not an identifier to a particular set of elements i.e. in this case `startInputGroup`.

Reply:
Oh, I see!
Ok, let it stay. I am a bit concerned we may be abusing the `role` attribute (which, eventually, has to do with accessibility), and if we end up really needing to identify otherwise indistinguishable elements in the DOM during tests, we will need to use `data-test-id`s (and a [babel plugin](https://www.npmjs.com/package/babel-plugin-jsx-remove-data-test-id) for their removal from production code).